BlueShift-iOS-SDK/BlueShiftAppDelegate.m

```diff
@@ -1245,4 +1245,37 @@ -(void)processUniversalLinks:(NSURL * _Nonnull)url {
     }
 }
 
+- (void)sceneDidBecomeActive:(UIScene* _Nullable)scene API_AVAILABLE(ios(13.0)) {
+    if (BlueShift.sharedInstance.config.isSceneDelegateConfiguration == YES) {
+        [self appDidBecomeActive:UIApplication.sharedApplication];
+    }
+}
+
+- (void)sceneDidEnterBackground:(UIScene* _Nullable)scene API_AVAILABLE(ios(13.0)) {
+    if (BlueShift.sharedInstance.config.isSceneDelegateConfiguration == YES) {
+        if ([NSThread isMainThread] == YES) {
+            [self processSceneDidEnterBackground];
+        } else {
+            dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
what happens if we call this from the main thread? if it will not break, is it good to have just this code here?

Reply:
New mainThread execution call is async, it adds the job in the main dispatch queue and the execution may get delayed. Most of times this functionality will get called on main thread. I have added the block just for precaution.